testutil/testnode/full_node.go

```diff
@@ -195,13 +200,17 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (cleanup func() error
 	tmNode, app, cctx, err := New(t, DefaultParams(), tmCfg, false, genState, kr)
 	require.NoError(t, err)
 
+	// locking the mutex not to be able to spin up multiple nodes at the same time.
+	mut.Lock()
```

Comment:
[blocking suggestion]
we should probably put this in the `New` function instead of the default, as the default is only a helper function, and sometimes we need to specify things outside of the default

Reply:
The `New` is used to prepare the node, however the `DefaultNetwork` starts it. I guess that we want the lock when we want to start the node. If a user uses `New` to create the node structure (without starting it), then it's up to him to handle the locks, no?

Reply:
we have other scenarios where we won't call default node, for example, each time we want to changes something in the config, consensus params, or genesis state. I think adding it it either the new or start function will accomplish the same thing, so which ever is fine imo

Reply:
[35c4042](https://github.com/celestiaorg/celestia-app/pull/1342/commits/35c4042788f56111ba1d1ca449f8045cf73f889f)  something like this?